waf/waf.go

```diff
@@ -3,21 +3,24 @@
 // Then add the rule to the Rules slice.
 package waf
 
-import "github.com/terraform-linters/tflint-plugin-sdk/tflint"
+import (
+	"reflect"
 
-// Rules is the list of rules for Well Architected Alignment.
-// Make sure to add any new rules to this list.
-// Please sort the list to be kind to your fellow maintainers.
-var Rules = []tflint.Rule{
-	AzurermApplicationGatewaySku(),
-	AzurermApplicationGatewayZones(),
-	AzurermKubernetesClusterZones(),
-	AzurermLbSku(),
-	AzurermPublicIpSku(),
-	AzurermPublicIpZones(),
-	AzurermServicePlanZoneBalancingEnabled(),
-	AzurermStorageAccountAccountReplicationType(),
-	AzurermVirtualMachineZoneUnknown(),
-	AzurermVirtualMachineZonesUnknown(),
-	AzurermVirtualNetworkGatewaySku(),
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
```

Comment:
Sorry to nit pick but can we add some docs to this type, and to the `GetRules` func below?


```suggestion
// WafRules is a helper struct. Methods are created on this type that generate the rules for the WAF package.
// We then use reflection to automatically generate a slice of the rules to add the the ruleset.
// This avoids having to manually maintain a list of rules.
// See `GetRules()` for more detail.
```